### PR TITLE
Manasa-Unlocked Screen: Edit Tiles

### PR DESCRIPTION
**Pull request by manasapakala191**
Added/Modified :
- Edit tile with colour switching, editable name, and other UI changes
- Delete dialog
- UI change on tiles screen after long press
- Image picker for edit screen
- New Tile/ Board
- Voice recorder and playback

https://user-images.githubusercontent.com/57900828/127508874-250e1478-dc41-4d96-af4b-a4425a135eb3.mp4

### CODE REVIEW

File: lib/unlocked/widgets/Tiles.dart [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                title: Text('Hide'),
+              ),
+              ListTile(
+                leading: Icon(
+                  Icons.delete,
+                  color: paua,
+                ),
+                title: Text(
+                  'Delete',
+                  style: TextStyle(color: Color(0xffC62828)),
```
Comment (tungtrn) [inline lib/unlocked/widgets/Tiles.dart:187]: This should be put in `Contant.dart` file

File: lib/unlocked/widgets/Tiles.dart [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                    height: 230.75,
+                    color: light_yellow,
+                    child: Image.asset(
+                      tile.imageUrl,
+                      width: 277.11,
+                    ),
+                  ),
+                ),
+                Container(
+                  height: 21.44,
```
Comment (tungtrn) [inline lib/unlocked/widgets/Tiles.dart:135]: Can we have the length value set relative to `screenSize` like other properties?

**Review by tungtrn — CHANGES_REQUESTED** [review]
Good jobb @manasapakala191 !

I checked out the screens and it is great! 

I added some minor comments about the constant and the screen size. Also, I merge `Summer2021` into your branch and resolve some conflicts. Please check if your changes are affected.

File: lib/data/data.dart [hunk 3]
```diff
@@ -1,38 +1,154 @@
...
+      name: 'Test',
+      isFile: true,
+      imageUrl: 'assets/images/test.png',
+      vocalization: 'Test Vocal',
+      voiceRecorder: true),
+  Tile(
+      name: 'Label',
+      editName: 'Please enter name',
+      isFile: true,
+      imageUrl: 'assets/images/empty.png',
```
Comment (tungtrn) [inline lib/data/data.dart:151]: I can't find `empty.png` so it causes the error when running

**Review by tungtrn — CHANGES_REQUESTED** [review]
@manasapakala191  Thank you so much for fixing the last comment:
There are 2 minor issues that I found:
- There isn't a file named `empty.png` in the asset so I could not find it. Did you forget to commit it?
- The image below is a bit different from the design, the color of the tiles should be fully covered
Screen Shot 2021-08-03 at 7 36 23 PM

Also, I added a .gitignore file in the `Summer2021` branch, it would be great if you can pull it to your branch so we don't have unnecessary files

File: lib/unlocked/EditTile.dart [hunk 4]
```diff
@@ -35,11 +37,89 @@ class _EditTileScreenState extends State<EditTileScreen> {
...
+                      height: 15,
+                    ),
+                    Text("This action cannot be undone.", style: TextStyle(color: Colors.grey),),
+                    SizedBox(
+                      height: 40,
+                    ),
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        OutlinedButton(
```
Comment (nganphan123) [inline lib/unlocked/EditTile.dart:99]: There is a `button` class in shared folder. I think you can try using that

File: lib/unlocked/EditTile.dart [hunk 5]
```diff
@@ -35,11 +37,89 @@ class _EditTileScreenState extends State<EditTileScreen> {
...
+                          autofocus: true,
+                          style: OutlinedButton.styleFrom(
```
Comment (nganphan123) [inline lib/unlocked/EditTile.dart:101]: Same as above

**Review by nganphan123 — COMMENTED** [review]
Great work Manasa. I just added a small comment.

**Comment by manasapakala191** [comment]
Updates:

- Select, select-all for edit, delete updated
- Navigation string to folder on the home screen
- Download the voice recording and store offline

Questions:
1. Should the sentence bar be added in the unlocked home screen as well?
2. In the edit/add tile page, do I add a test vocal button for testing the TTS for the tile label?

**Review by tungtrn — APPROVED** [review]
Hi Rukmini,

Thank you so much for fixing the issues and implementing the UI.

The functionalities and the UI looks good on my end. There are some minor places that the UI is overflowed on my screen. I guess it may be because of the fixed padding size and you may want to consider that (Screenshot attached below)
Screen Shot 2021-09-25 at 11 04 43 PM


Since this is not a major issue, you can work on it while you are integrating the API as well so I will approve it for now.

Thank you so much!! 👏👏

Please let me know if you have any questions!

**Pull request merged**